src/field/arena.ts

```diff
@@ -643,6 +643,25 @@ export class Arena {
     return TimeOfDay.DAWN;
   }
 
+  /**
+   * Sets a random weather based on the time of day and the current biome
+   */
+  setRandomWeather(): void {
+    const weatherPool = allBiomes.get(this.biomeId).weatherPool;
+    const weatherMap = new Map<WeatherType, number>();
+    for (const id of getEnumValues(WeatherType)) {
+      weatherMap.set(id, weatherPool[id]);
```

Comment:
```suggestion
      weatherMap.set(id, weatherPool[id] ?? 0);
```
If Pigeon's suggestion to make the weather pool field a `Partial` record is used I think you'll also need to do this.